### DISCUSSION TITLE
Where is SpeechRecognizer?

### DISCUSSION THREAD

**Discussion by sej69** [dotnet/maui#14220] [Q&A]
I see the SpeechOptions is available in Microsoft.Maui.Media, but I don't see the method that actually uses that class.  It should be called "SpeechRecognizer", or at least that is what it's called in Xamarin.  Any ideas where I can find this in the new libraries that won't be deprecated soon?

**Comment by jfversluis** [comment]
I'm not sure what you mean. Can you link me to something that was available in Xamarin?

I don't remember any SpeechRecognizer being part of Xamarin.Forms or Xamarin.Essentials. If you mean the APIs that are available on the platform then you should be able to find it whenever you write code in one of the subfolders for each platform in the Platforms folder.

If you have some more info on where you were able to find it for Xamarin or what you mean exactly I might be able to help you further.

**Reply by sej69** [reply]
Yes, the speechrecognizer was in in Xamarin.Essentials.  In all my reading, it appears Xamarin.Essentials is being deprecated and replaced with this project.  This is the section here: https://learn.microsoft.com/en-us/xamarin/xamarin-forms/data-cloud/azure-cognitive-services/speech-recognition  Specifically, the speech to text (speechrecognizer) is what I am looking for.

**Reply by sej69** [reply]
Any word on this? I'm stuck in limbo at the moment.  I'm half converted over to the new API but am stuck without the speech recognition.

**Reply by xtuzy** [reply]
You must remember it wrong, speechrecognizer not in essential , 
essential only have text to speech, see this list: https://learn.microsoft.com/en-us/xamarin/essentials/?context=xamarin%2Fxamarin-forms. 
speechrecognizer is a service of azure, it is not free.

**Reply by sej69** [reply]
@xtuzy  I'm baffled by your response; I'm looking at my Xamarin code right now and have never used Azure for this.  In fact, my app is not connected to the internet frequently while in use; it would never work in this scenario.  I did look at Azure recently, but I don't need full speech recognition, I'm fine with pattern constraints.  And the fact that I need to be connected to the internet with Azure makes it unusable for this scenario.  It looks like I did link the wrong URL though, I've been scrambling to find the updated libraries.

**Reply by xtuzy** [reply]
if you want native SpeechRecognition, see: https://github.com/shinyorg/shiny/tree/master/src/Shiny.SpeechRecognition
